setup.sh

```diff
@@ -318,11 +318,12 @@ setup_arc() {
 }
 
 
+install_dotfiles
```

Comment:
does it become problematic to install the dotfiles before checking dependencies? I was thinking we had to do the check first, but maybe not?

Reply:
It is possible that the git check in check_dependencies might find an issue impacting install_dotfiles, but it'd be rare and is a bit circular. 

However, on M1, we absolutely need the dotfiles installed before the npm check in check_dependencies.